src/LibVLCSharp.Forms/Shared/MediaPlayerElement.xaml.cs

```diff
@@ -242,7 +259,189 @@ private void PageDisappearing(object sender, Page e)
+                AmpBands = new ObservableCollection<Band>(CopyBands(SelectedPreset.Bands));
+                PresetsDataPicker.SelectedIndex = SelectedPreset.PresetId;
+
+                PreampSlider.BindingContext = SelectedPreset;
+                PreampSlider.SetBinding(Slider.ValueProperty, nameof(SelectedPreset.Preamp));
+
+                if (FrequenciesLayout != null)
+                    FrequenciesLayout.ItemsSource = AmpBands;
+            }
+        }
+
+        private void BandAmplificationValueChanged(object sender, ValueChangedEventArgs e)
+        {
+            EnableEqualizerIfDisable();
+            if (!EqualizerViewIsOpening && !IgnoreAmpValueChanged && !IsTheSameAmplification(e.OldValue, e.NewValue))
+            {
+                var ampSwitch = (Slider)sender;
+                // Get the current band from where the Slider value was updated
+                var mainView = (View)ampSwitch.Parent.Parent;
+                var band = (Band)mainView.BindingContext;
+                
+                if(SelectedPreset != null  && MediaEqualizer != null && SnapBandsSwitch != null)
+                {
+                    // Snap Band is enable => Perform frequency smoothing
+                    if (SnapBandsSwitch.IsToggled && FrequenciesLayout != null)
+                    {
+                        AmpSliderTriggerCount++;
+                        AmpSliderTriggerCount = AmpSliderTriggerCount >= 10 ? 0 : AmpSliderTriggerCount;
+                        if (AmpSliderTriggerCount == 1)
+                        {
+                            IgnoreAmpValueChanged = true;
+                            SmoothOutFrequencies(band.BandId, (float)e.OldValue,
+                                (float)e.NewValue, AmpBands, MediaEqualizer);
+                            SetEqualizerToMediaPlyer(MediaPlayer, MediaEqualizer);
+                            IgnoreAmpValueChanged = false;
+                        }
+                    }
+                    else
+                    {
+                        ApplyNewAmplification(band.BandId, (float)e.NewValue, MediaEqualizer);
+                        SetEqualizerToMediaPlyer(MediaPlayer, MediaEqualizer);
+                    }
+                }
+            }
+        }
+
+        private void EnableEqualizerSwitchToggled(object sender, ToggledEventArgs e)
+        {
+           if(e.Value)
+           {      
+                if (SelectedPreset != null && MediaEqualizer != null && !EqualizerViewIsOpening)
+                {
+                    MediaEqualizer = new Equalizer((uint)SelectedPreset.PresetId);
+                    MediaEqualizer.SetPreamp((float)SelectedPreset.Preamp);
+
+                    SaveEqualizerState(true);
+                    SavePreset(SelectedPreset.PresetId);
+                    SetEqualizerToMediaPlyer(MediaPlayer, MediaEqualizer);
+                }
+           }
+           else
+           {
+                MediaPlayer.UnsetEqualizer();
```

Comment:
do we want to check the return value for this?

Reply:
Not sure we need to check the returned value. I say that because if the returned value is false for example, there is nothing to do.

Reply:
ok